lib/rules/import.js

```diff
@@ -31,7 +35,42 @@ module.exports = {
                     .forEach((imp) => {
                         context.report({ node, message: imp.getErrMessage() });
                     });
-            }
+            },
+            MemberExpression(node) {
+                if ((node.object.callee.name !== 'require' || !node.object.arguments.length) && !node.property) {
+                    return;
+                }
+                const requireArg = node.object.arguments[0];
+                if (requireArg.type !== 'Literal') {
+                    return;
+                }
+                imports
+                    .filter((imp) => imp.check(requireArg.value, new Set([node.property.name])))
+                    .forEach((imp) => {
+                        context.report({ node, message: imp.getErrMessage() });
+                    });
+            },
+            VariableDeclaration(node) {
+                if (!(node.declarations && node.declarations.length)) return;
+
+                const childNode = node.declarations[0];
+                if (!(childNode.id && childNode.id.type === 'ObjectPattern' && childNode.init)) return;
+
+                const requireArg = childNode.init.arguments[0];
+                if (requireArg.type !== 'Literal' || childNode.init.callee.name !== 'require') {
+                    return;
+                }
+
+                const namedImports = childNode.id.properties
+                  .filter((module) => !!module.value.name)
+                  .map((module) => module.value.name);
+
+                imports
+                    .filter((imp) => imp.check(requireArg.value, new Set(namedImports)))
```

Comment:
creating Set from array is O(n) operation, better to use it outside of the filter func